src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -323,6 +332,54 @@ void CollectError(Type extensionType, Version minimumVersion, ExtensionStartupTy
             }
         }
 
+        internal static bool IsValidBindingMatch(ExtensionReference extension, HashSet<string> bindingsSet, IEnvironment environment)
+        {
+            if (extension.Bindings.Count == 0)
+            {
+                return true;
+            }
+
+            // Application Insights uses a special binding value to ensure it's installed based on its App Setting
+            if (extension.Bindings.Any(b => b.Equals("_")))
+            {
+                return IsApplicationInsightsSettingPresent(environment);
+            }
+
+            return extension.Bindings.Intersect(bindingsSet, StringComparer.OrdinalIgnoreCase).Any();
+        }
+
+        internal static void ValidateApplicationInsightsConfig(List<Type> startupTypes, bool isSelfHost, bool isApplicationInsightsSettingPresent, bool isExtensionBundleConfigured, ILogger logger)
+        {
+            if (isSelfHost)
+            {
+                logger.LogWarning("In order to use Application Insights in Azure Functions V4 and above, please install the Application Insights Extension." +
+                                  "See https://aka.ms/func-applicationinsights-extension for more details.");
+            }
+            else if (!isExtensionBundleConfigured)
```

Comment:
I did this because in both cases, extensionbundles shouldn't be involved. If it's installed but there's no app setting, we shouldn't throw an exception because the app setting is being checked earlier in `IsValidBindingMatch` which is only called for bundles. If it's not installed and there's an app setting, I left a comment on why that could be problematic.